.github/workflows/build.yml

```diff
@@ -149,6 +149,33 @@ jobs:
           . ../firedrake_venv/bin/activate
           firedrake-run-split-tests 8 1 "$EXTRA_PYTEST_ARGS --junit-xml=firedrake8_{#}.xml"
 
+      - name: Test pyadjoint
+        if: (success() || steps.build.conclusion == 'success') && matrix.scalar-type == 'real'
+        run: |
+          . ../firedrake_venv/bin/activate
+          cd ../firedrake_venv/src/pyadjoint
+          python -m pytest \
+            --strict-markers \
+            --durations=200 \
+            --timeout=600 \
+            --timeout-method=thread \
+            -o faulthandler_timeout=660 \
+            -n 12 --dist worksteal \
+            -sv tests/firedrake_adjoint
+        timeout-minutes: 5
+
+      - name: Run Gusto smoke tests
+        # NOTE: Unsure if Gusto is expected to work in complex mode
+        if: (success() || steps.build.conclusion == 'success') && matrix.scalar-type == 'real'
+        run: |
+          . ../firedrake_venv/bin/activate
+          cd ../firedrake_venv/src/gusto
+          python -m pytest -n 12 -v \
```

Comment:
Are these all serial?

Reply:
I don't think we need to specify those things here. There is a GitHub-enforced limit of 5 minutes for the test and we don't care how long each test takes to run.

And they are all serial.